### PR TITLE
Target .NET Standard 2.0

### PR DESCRIPTION
**Pull request by natemcmaster**
Changes

 - lifts from netstandard1.3 to netstandard2.0
 - added net46 where required to ensure we don't need netstandard.dll in the VSIX
 - updated corefx version (only affects one package in test, System.Diagonstics.DiagnosticsSource)
 - float NS.Library correctly to avoid package downgrade warnings

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -18,7 +18,11 @@
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)'=='.NETFramework' AND '$(OutputType)'=='library'">
-    <PackageReference Include="NETStandard.Library" Version="$(BundledNETStandardPackageVersion)" />
+    <PackageReference Include="NETStandard.Library" Version="$(NETStandardImplicitPackageVersion)" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)'=='.NETFramework'">
+    <PackageReference Include="NETStandard.Library.NETFramework" Version="$(NETStandardLibraryNETFrameworkVersion)" PrivateAssets="All" />
```
Comment (rynowak) [inline build/common.props]: Is this still going to add the `netstandard.dll` to projects like `Microsoft.AspNetCore.Razor.Language`? I might be confused, can you explain?

File: test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Razor.Test.MvcShim.csproj [hunk 2]
```diff
@@ -3,13 +3,13 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFrameworks>net46;netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks>net461;netcoreapp2.0</TargetFrameworks>
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Razor.Test.MvcShim.csproj]: Why the change here? I'd prefer to keep these in sync with the src projects, but it's not a huge deal if there's a good reason why we need to lift it.

**Comment by rynowak** [comment]
See https://github.com/aspnet/Razor/blob/dev/tooling/Microsoft.VisualStudio.RazorExtension/Microsoft.VisualStudio.RazorExtension.csproj#L252

I think you're probably ok since all of these still build as net46.

**Comment by rynowak** [comment]
The other thing to check is, if you build the VSIX and then unzip it, are they any extraneous assemblies in there. This is really easy to regress.

File: build/common.props (hunk 1 above)
Comment (natemcmaster) [inline build/common.props]: Technically it shouldn't, but to be safe, I'll move this package ref out of common and add it only to csproj's were we are sure it's ok.

File: test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Razor.Test.MvcShim.csproj (hunk 2 above)
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Razor.Test.MvcShim.csproj]: This test project depends on Razor.Runtime, which is ns2.0 only so I had to lift to net461 to make it compatible.

**Comment by natemcmaster** [comment]
VSIX contents look ok. Plus, I verified manually that none of these assemblies were compiled with a reference to netstandard.dll.

image

File: test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Razor.Test.MvcShim.csproj (hunk 2 above)
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Test.MvcShim/Microsoft.AspNetCore.Razor.Test.MvcShim.csproj]: Ah right! 👍
